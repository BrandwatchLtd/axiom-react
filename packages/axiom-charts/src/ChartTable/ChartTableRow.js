import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Base } from '@brandwatch/axiom-components';

export default class ChartTableRow extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  render() {
    const { children, ...rest } = this.props;

    return (
      <Base { ...rest } className="ax-chart-table__row">
        { children }
      </Base>
    );
  }
}
