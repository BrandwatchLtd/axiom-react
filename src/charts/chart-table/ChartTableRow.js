import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Base } from 'bw-axiom';

export default class ChartTableRow extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  render() {
    const { children, ...rest } = this.props;

    return (
      <Base { ...rest } className="ax-chart-table__row">
        { children }
      </Base>
    );
  }
}
