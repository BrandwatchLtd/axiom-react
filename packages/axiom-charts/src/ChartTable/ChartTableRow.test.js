import React from 'react';
import renderer from 'react-test-renderer';
import ChartTableRow from './ChartTableRow';

const getComponent = (props = {}) =>
  renderer.create(
    <ChartTableRow { ...props }>
      Lorem Ipsum
    </ChartTableRow>
  );

describe('ChartTableRow', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
