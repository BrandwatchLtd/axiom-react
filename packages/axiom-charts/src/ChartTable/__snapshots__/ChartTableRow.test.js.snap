// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartTableRow renders with defaultProps 1`] = `
<div
  className="ax-chart-table__row"
>
  Lorem Ipsum
</div>
`;
